modules/UKBiobank.py

```diff
@@ -0,0 +1,201 @@
+                        evidenceString.disease = bioentity.Disease(
+                            id=disease_id,
+                            #TODO update disease label
+                            name=disease_id
+                        )
+
+                        '''
+                            build variant object
+                        '''
+                        evidenceString.variant = bioentity.Variant(
+                            id=variant_id,
+                            type="snp single"
+                        )
+
+                        '''
+                            build evidence variant2disease object
+                        '''
+                        #TODO variant2disease & gene2variant object missing
+                        evidenceString.evidence = opentargets.GeneticsEvidence(
+                            variant2disease=evidence_genetics.Variant2Disease(
+                                unique_experiment_reference=study,
+                                date_asserted=now.isoformat(),
+                                resource_score=resource_score,#evidence_score.Pvalue(value=2.000000039082963e-25),
+                                provenance_type=provenance_type,
+                                evidence_codes=["http://purl.obolibrary.org/obo/ECO_0000205"],
+                                gwas_sample_size=int(sample_size)
+                            ),
+                            gene2variant=evidence_genetics.Gene2Variant(
+                                date_asserted=now.isoformat(),
+                                provenance_type=provenance_type,
+                                evidence_codes=["http://purl.obolibrary.org/obo/ECO_0000205"],
+                                functional_consequence=func_conseq
+                            ),
+                        )
+                        error = evidenceString.validate(logging)
+
+                        if error > 0:
+                            self.logger.error(evidenceString.to_JSON())
+                            sys.exit(1)
+
+                        self.evidence_strings.append(evidenceString)
+
+                    else:
+                        self.logger.error("%s is not found in Ensembl" % gene_symbol)
+
+            self.logger.info("%s evidence parsed"%(n-1))
+            self.logger.info("%s evidence created"%len(self.evidence_strings))
+
+        ukbiobank_input.close()
+
+    def write_evidence(self, filename=Config.UKBIOBANK_EVIDENCE_FILENAME):
+        self.logger.info("Writing UKBiobank evidence strings")
+        with open(filename, 'w') as ukbiobank_output:
+            n = 0
+            for evidence_string in self.evidence_strings:
+                n += 1
+                self.logger.info(evidence_string.disease.id[0])
+                # get max_phase_for_all_diseases
+                error = evidence_string.validate(logging)
+
+                if error == 0:
+                    print(evidence_string.to_JSON(indentation=None))
```

Comment:
please, logging from the beginning makes the code better and more readable and debuggable